crates/resolver/src/name_server_pool.rs

```diff
@@ -901,19 +914,35 @@ where
                                 request_loop,
                                 ProtoError::from("No connections available"),
                             ),
-                            |(mut conns, request, err)| {
-                                let conn = conns.pop();
-
+                            move |(mut conns, request, err)| {
                                 let request_cont = request.clone();
 
-                                conn.ok_or_else(move || err).into_future().and_then(
-                                    move |mut conn| {
-                                        conn.send(request).then(|sent_res| match sent_res {
-                                            Ok(sent) => Ok(Loop::Break(sent)),
-                                            Err(err) => {
+                                // construct the parallel requests, 2 is the default
+                                let mut par_conns = SmallVec::<[NameServer<C, P>; 2]>::new();
+                                let count = conns.len().min(opts.num_concurrent_reqs);
```

Comment:
I need to enforce this is a min of 1, which I missed. I'll patch that, will be a good thing to test for.

Reply:
Fix, and the checked in tests, did catch this.